src/Illuminate/Support/Collection.php

```diff
@@ -1910,6 +1910,10 @@ public function pad($size, $value)
     public function toArray()
     {
         return array_map(function ($value) {
+            if ($value instanceof \stdClass) {
+                return (array) $value;
```

Comment:
When I've a `class Foo extends stdClass extends Arrayable` it gets interesting.

I think in this case the `Arrayable` check should be first.

Reply:
Do you mean `class Foo extends stdClass implements Arrayable`?

Yeah that makes sense. I'll make the change.